ArcGISRuntimeSDKQt_CppSamples/EditData/UpdateAttributesFeatureService/UpdateAttributesFeatureService.cpp

```diff
@@ -164,21 +177,32 @@ void UpdateAttributesFeatureService::connectSignals()
 
 void UpdateAttributesFeatureService::updateSelectedFeature(QString fieldVal)
 {
-  // connect to load status changed signal
-  connect(m_selectedFeature, &ArcGISFeature::loadStatusChanged,
-          this, [this, fieldVal](Esri::ArcGISRuntime::LoadStatus)
-  {
-    if (m_selectedFeature->loadStatus() == LoadStatus::Loaded)
-    {
-      disconnect(m_selectedFeature, &ArcGISFeature::loadStatusChanged, 0, 0); // bad...
-
-      // update the select feature's attribute value
-      m_selectedFeature->attributes()->replaceAttribute("typdamage", fieldVal);
+  // Shared connection object that we will use to disconnect the connection in the slot.
+  // We do this to create a form of fire-and-forget where the slot can only ever be invoked once.
+  QSharedPointer<QMetaObject::Connection> connection { QSharedPointer<QMetaObject::Connection>::create() };
 
-      // update the feature in the feature table
-      m_featureTable->updateFeature(m_selectedFeature);
-    }
-  });
+  // connect to load status changed signal
+  QMetaObject::Connection c =
+      connect(
+          m_selectedFeature, &ArcGISFeature::loadStatusChanged,
+          this, [this, fieldVal, connection](Esri::ArcGISRuntime::LoadStatus)
+                {
+                  if (m_selectedFeature->loadStatus() == LoadStatus::Loaded)
+                  {
+                    // The conenction is invoked so we now forget all about this connection after this point.
+                    disconnect(*connection);
+
+                    // update the select feature's attribute value
+                    m_selectedFeature->attributes()->replaceAttribute("typdamage", fieldVal);
+
+                    // update the feature in the feature table
+                    m_featureTable->updateFeature(m_selectedFeature);
+                  }
+                }
+  );
+
+  // Copy out the connection data to the shared connection.
+  *connection = c;
```

Comment:
Can this connection be a class member? There's a lot of work going on here to prevent shared_ptr cycles with the connection and lambda capture.